### PR TITLE
 python3.pkgs.django-postgresql-netfields: run tests & update

### PR DESCRIPTION
**Pull request by yu-re-ka**
## Description of changes

https://github.com/jimfunk/django-postgresql-netfields/blob/v1.3.1/CHANGELOG

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yu-re-ka** [comment]
Built nixosTests.peering-manager

**Comment by mweinelt** [comment]
Please add `meta.changelog`.

https://github.com/jimfunk/django-postgresql-netfields/blob/v1.3.1/CHANGELOG

**Comment by yu-re-ka** [comment]
`meta.changelog` is already set. do you mean I should add the link to the changelog for this version to the commit message?

**Comment by mweinelt** [comment]
Ah, I missed it was already set. But yeah, changelogs should be referenced in the commit message according to CONTRIBUTING.md.

**Comment by mweinelt** [comment]
@ofborg build python311.pkgs.django-postgresql-netfields

File: pkgs/development/python-modules/django-postgresql-netfields/default.nix [hunk 1]
```diff
@@ -30,23 +34,17 @@ buildPythonPackage rec {
     six
   ];
 
-  # tests need a postgres database
-  doCheck = false;
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-postgresql-netfields/default.nix:34]: The tests can only run on Linux if they need postgresql.

```

django-postgresql-netfields-aarch64-darwin> Executing pytestCheckPhase
django-postgresql-netfields-aarch64-darwin> initializing postgresql
django-postgresql-netfields-aarch64-darwin> The files belonging to this database system will be owned by user "_nixbld1".
django-postgresql-netfields-aarch64-darwin> This user must also own the server process.
django-postgresql-netfields-aarch64-darwin> 
django-postgresql-netfields-aarch64-darwin> The database cluster will be initialized with locale "en_US.UTF-8".
django-postgresql-netfields-aarch64-darwin> The default database encoding has accordingly been set to "UTF8".
django-postgresql-netfields-aarch64-darwin> The default text search configuration will be set to "english".
django-postgresql-netfields-aarch64-darwin> 
django-postgresql-netfields-aarch64-darwin> Data page checksums are disabled.
django-postgresql-netfields-aarch64-darwin> 
django-postgresql-netfields-aarch64-darwin> creating directory /private/tmp/nix-build-python3.10-django-postgresql-netfields-1.3.1.drv-0/postgresql ... ok
django-postgresql-netfields-aarch64-darwin> creating subdirectories ... ok
django-postgresql-netfields-aarch64-darwin> selecting dynamic shared memory implementation ... posix
django-postgresql-netfields-aarch64-darwin> selecting default max_connections ... 20
django-postgresql-netfields-aarch64-darwin> selecting default shared_buffers ... 400kB
django-postgresql-netfields-aarch64-darwin> selecting default time zone ... UTC
django-postgresql-netfields-aarch64-darwin> creating configuration files ... ok
django-postgresql-netfields-aarch64-darwin> running bootstrap script ... 2023-09-25 20:14:20.055 UTC [87086] FATAL:  could not create shared memory segment: Operation not permitted
django-postgresql-netfields-aarch64-darwin> 2023-09-25 20:14:20.055 UTC [87086] DETAIL:  Failed system call was shmget(key=371780822, size=56, 03600).
django-postgresql-netfields-aarch64-darwin> child process exited with exit code 1
django-postgresql-netfields-aarch64-darwin> initdb: removing data directory "/private/tmp/nix-build-python3.10-django-postgresql-netfields-1.3.1.drv-0/postgresql"

**Comment by yu-re-ka** [comment]
I'm confused: the build passed in ofborg x86_64-darwin even before I disabled the test: https://github.com/NixOS/nixpkgs/runs/17115565582

**Comment by mweinelt** [comment]
Huh, interesting. Feel free to reenable them on x86_64-darwin then.

**Comment by yu-re-ka** [comment]
I don't feel competent to make a decision on that (does ofborg have sandboxing enabled? is this an x86_64-darwin vs aarch64-darwin thing?), so I would leave them disabled for now

**Comment by mweinelt** [comment]
Yes, best to leave it off on darwin. I can reproduce the error in x86_64-darwin as well. Remember to accept `stdenv` in the function arguments, currently `stdenv.isDarwin` does not resolve.

```
django-postgresql-netfields-x86_64-darwin> Executing pytestCheckPhase
django-postgresql-netfields-x86_64-darwin> initializing postgresql
django-postgresql-netfields-x86_64-darwin> The files belonging to this database system will be owned by user "_nixbld1".
django-postgresql-netfields-x86_64-darwin> This user must also own the server process.
django-postgresql-netfields-x86_64-darwin> 
django-postgresql-netfields-x86_64-darwin> The database cluster will be initialized with locale "en_US.UTF-8".
django-postgresql-netfields-x86_64-darwin> The default database encoding has accordingly been set to "UTF8".
django-postgresql-netfields-x86_64-darwin> The default text search configuration will be set to "english".
django-postgresql-netfields-x86_64-darwin> 
django-postgresql-netfields-x86_64-darwin> Data page checksums are disabled.
django-postgresql-netfields-x86_64-darwin> 
django-postgresql-netfields-x86_64-darwin> creating directory /private/tmp/nix-build-python3.10-django-postgresql-netfields-1.3.1.drv-0/postgresql ... ok
django-postgresql-netfields-x86_64-darwin> creating subdirectories ... ok
django-postgresql-netfields-x86_64-darwin> selecting dynamic shared memory implementation ... posix
django-postgresql-netfields-x86_64-darwin> selecting default max_connections ... 20
django-postgresql-netfields-x86_64-darwin> selecting default shared_buffers ... 400kB
django-postgresql-netfields-x86_64-darwin> selecting default time zone ... UTC
django-postgresql-netfields-x86_64-darwin> creating configuration files ... ok
django-postgresql-netfields-x86_64-darwin> running bootstrap script ... 2023-09-25 20:51:21.592 UTC [87935] FATAL:  could not create shared memory segment: Operation not permitted
django-postgresql-netfields-x86_64-darwin> 2023-09-25 20:51:21.592 UTC [87935] DETAIL:  Failed system call was shmget(key=371781163, size=56, 03600).
django-postgresql-netfields-x86_64-darwin> child process exited with exit code 1
django-postgresql-netfields-x86_64-darwin> initdb: removing data directory "/private/tmp/nix-build-python3.10-django-postgresql-netfields-1.3.1.drv-0/postgresql"

**Pull request merged**